pinc/graph_data.inc

```diff
@@ -289,3 +289,212 @@ function users_by_country()
+    $result = DPDatabase::query($sql);
+
+    [$datax, $y_cumulative] = dpsql_fetch_columns($result);
+
+    $datay1 = array_successive_differences($y_cumulative);
+
+    // Pad out the rest of the month
+    for ($i = count($datay1); $i < $maxday; $i++) {
+        $datax[$i] = $i + 1;
+        $datay1[$i] = 0;
+    }
+
+    return [
+        "title" => "$psd->per_day_title ($timeframe)",
+        "yAxisWidth" => 75,
+        "data" => [
+            _('Projects') => [
+                "x" => $datax,
+                "y" => $datay1,
+            ],
+        ],
+    ];
+}
+
+function array_subtract_first_from_each($arr)
+// Return the array defined by:
+//     $result[$i] = $arr[$i] - $arr[0]
+{
+    $result = [];
+    foreach ($arr as $key => $value) {
+        if (count($result) == 0) {
+            $first_value = $value;
+        }
+        $result[$key] = $value - $first_value;
+    }
+    return $result;
+}
+
+function cumulative_month_proj($which)
+{
+    // Create "projects Xed per day" graph for current month
+
+    $psd = get_project_status_descriptor($which);
+
+    $todaysTimeStamp = time();
+
+    $year = date("Y", $todaysTimeStamp);
+    $month = date("m", $todaysTimeStamp);
+    $monthVar = _(date("F", $todaysTimeStamp));
+    $timeframe = "$monthVar $year";
+
+    $maxday = get_number_of_days_in_current_month();
+
+    //query db and put results into arrays
+    $sql = sprintf("SELECT DAYOFMONTH(date) as day, SUM(num_projects)
+        FROM project_state_stats
+        WHERE MONTH(date) = %s AND YEAR(date) = %s AND (%s)
+        GROUP BY DAYOFMONTH(date)
+        ORDER BY date",
+        DPDatabase::escape($month),
+        DPDatabase::escape($year),
+        $psd->state_selector);
+    $result = DPDatabase::query($sql);
+
+    [$datax, $y_num_projects] = dpsql_fetch_columns($result);
+
+    // get base level, total at beginning of 1st day of month
+    // snapshot is taken just after midnight,
+    // so day = 1 has total at beginning of month
+    // Subtract that base level from each subsequent day's value
+    $datay1 = array_subtract_first_from_each($y_num_projects);
+    array_shift($datay1);
+
+    // Pad out the rest of the month
+    for ($i = count($datay1); $i < $maxday; $i++) {
+        $datax[$i] = $i + 1;
+        $datay1[$i] = "";
+    }
+
+    return [
+        "title" => "$psd->cumulative_title ($timeframe)",
+        "yAxisWidth" => 75,
+        "data" => [
+            _('Projects') => [
+                "x" => $datax,
+                "y" => $datay1,
+            ],
+        ],
+    ];
+}
+
+function total_proj_graph($which)
+{
+    //Create "projects Xed per day" graph for all known history
+
+    $psd = get_project_status_descriptor($which);
+
+    $timeframe = _('since stats began');
+
+
+    //query db and put results into arrays
+    $result = DPDatabase::query("SELECT date, SUM(num_projects)
+      FROM project_state_stats
+      WHERE $psd->state_selector
+      GROUP BY date
+      ORDER BY date");
```

Comment:
Should this and the same style at line 477 be updated to use the updated querying styling?

Reply:
Yup, my search was faulty and only looked at $sql lines